tests/e2e.rs

```diff
@@ -737,17 +768,29 @@ impl Ldap {
 			.map(|(attribute, changes)| Mod::Replace(attribute, changes))
 			.collect();
 
+		let base_dn = if let Some(ref source_ldap) = config().await.source_ldap {
+			source_ldap.base_dn.as_str()
+		} else {
+			panic!("ldap must be configured for this test");
+		};
+
 		self.client
-			.modify(&format!("uid={},{}", uid, config().await.ldap.base_dn.as_str()), mods)
+			.modify(&format!("uid={},{}", uid, base_dn), mods)
 			.await
 			.expect("failed to modify user")
 			.success()
 			.expect("failed to modify user");
 	}
 
 	async fn delete_user(&mut self, uid: &str) {
+		let base_dn = if let Some(ref source_ldap) = config().await.source_ldap {
+			source_ldap.base_dn.as_str()
+		} else {
+			panic!("ldap must be configured for this test");
```

Comment:
Instead of a `let else` we can just use `.expect()` and `.map()`. Ditto for all the other places in the tests where we do this.